src/clients/c++/request_http.cc

```diff
@@ -1598,6 +1598,22 @@ InferHttpContextImpl::PreRunProcessing(std::shared_ptr<Request>& request)
     }
   }
 
+  for (const auto& io : outputs_) {
```

Comment:
Output::SetSharedMemory needs to move from Output to Options::AddSharedMemoryResult. This is consistent with how we handle other results. Then this code to initialize infer_request_.outputs should be put in SetRunOptions.

Reply:
Done